### PR TITLE
SNOW-1011764: Adding image-repository to supported objects.

### PR DESCRIPTION
**Pull request by sfc-gh-davwang**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Added image-repository to the supported objects for `snow object`. Note that DESCRIBE IMAGE REPOSITORY currently doesn't exist in SQL, thus `snow object describe` explicitly excludes image-repository for alignment.

Sample Output:
object drop image-repo
object list image-repo

### CODE REVIEW

**Comment by sfc-gh-turbaszek** [comment]
Please add integration tests

File: src/snowflake/cli/api/constants.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ class ObjectType(Enum):
     USER = ObjectNames("user", "user", "users")
     WAREHOUSE = ObjectNames("warehouse", "warehouse", "warehouses")
     VIEW = ObjectNames("view", "view", "views")
+    IMAGE_REPOSITORY = ObjectNames(
+        "image-repository", "image repository", "image repositories"
+    )
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/api/constants.py:43]:
```
@app.command(
    help=f"Provides description of an object of given type. {SUPPORTED_TYPES_MSG}"
)
@with_output
@global_options_with_connection
def describe(
```

Help still will be printing image repository as supported object for describe. You should change list of objects in help.

**Review by sfc-gh-yiwang — APPROVED** [review]
LGTM

File: tests/object/__snapshots__/test_object.ambr [hunk 2]
```diff
@@ -227,6 +227,14 @@
   
   '''
 # ---
+# name: test_describe_fails_image_repository
+  '''
+  ╭─ Error ──────────────────────────────────────────────────────────────────────╮
+  │ Object of type image_repository is not supported.                            │
```
[the remark this answers is not in the document]
Comment (sfc-gh-davwang) [reply tests/object/__snapshots__/test_object.ambr:233]: good spot, I've updated the test snapshot

**Pull request merged**